### DISCUSSION TITLE
createMany with relation

### DISCUSSION THREAD

**Discussion by SakoMe** [prisma/prisma#16557] [Q&A]
Hello looking for some guidance:

Two models:

```
model Terminal {
    terminal_id   Int      @id @default(autoincrement())
    terminal_name String
    created_at    DateTime @default(now())
    updated_at    DateTime @updatedAt
    modcods       Modcod[]

    @@map("terminal")
}
```

```
model Modcod {
    modcod_id   Int      @id @default(autoincrement())
    name        String?
    created_at  DateTime @default(now())
    updated_at  DateTime @updatedAt
    terminal    Terminal @relation(fields: [terminal_id], references: [terminal_id], onDelete: Cascade, onUpdate: Cascade)
    terminal_id Int

    @@map("modcod")
}
```

Standard create a singe resource works just fine:

```
export async function createSingleModcod(modcod: Modcod, terminal_id: number) {
  const modcod = await prisma.modcod.create({
    data: {
      ...modcod,
      terminal_id,
    },
  });
  await prisma.$disconnect();
  return modcod;
}
```
The create many (including using $transaction) does not:

```
export async function createManyModcods(modcods: Modcods[], terminal_id: number) {
  const modcods = await prisma.$transaction(
    modcods.map((modcod) =>
      prisma.modcod.create({
        data: {
          ...modcod,
          terminal_id, // this is the terminal_id coming from req.body
        },
      }),
    ),
  );
  await prisma.$disconnect();
  return modcods;
}
```

Can someone please let me know if I'm missing something or point me in the right direction?

**Comment by pantharshit00** [comment]
Hello @SakoMe,

Can you please tell what error you are encountering when you are running the second snippet?

**Reply by vladi-stevanovic** [reply]
👋 Hi @SakoMe, did you perhaps have a chance to check Harshit's question? Let us know if you still need help!

Or, if you've found a solution, we'd love to know how you solved this problem. Feel free to post it in this thread and "Mark as answer" if this discussion is now solved!

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
